pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroRecordExtractor.java

```diff
@@ -107,4 +120,23 @@ protected Object convertRecord(Object value) {
     }
     return convertedMap;
   }
+
+  /**
+   * This method convert any Avro logical-type converted (or not) value to a class supported by
+   * Pinot {@link GenericRow}
+   */
+  @Override
+  protected Object convertSingleValue(Object value) {
+    if (value instanceof BigDecimal) {
+      return Double.valueOf(((BigDecimal) value).doubleValue());
+    }
+    if (value instanceof LocalDate) {
+      return Long.valueOf(((LocalDate) value).toEpochDay());
+    }
+    if (value instanceof Instant) {
+      return Long.valueOf(((Instant) value).toEpochMilli());
+    }
```

Comment:
(minor) Can be simplified
```suggestion
    if (value instanceof BigDecimal) {
      return ((BigDecimal) value).doubleValue();
    }
    if (value instanceof LocalDate) {
      return ((LocalDate) value).toEpochDay();
    }
    if (value instanceof Instant) {
      return ((Instant) value).toEpochMilli();
    }
```